src/patterns/BidSnapshot/BidSnapshot.tsx

```diff
@@ -0,0 +1,147 @@
+
+import { getCommonProps } from '../../utils';
+import { DetailList } from '../DetailList/index';
+import { Detail } from '../../components/Detail/index';
+import { AuctionStatus, SupportedLanguages } from '../../types/commonTypes';
+import { Countdown } from '../../components/Countdown/index';
+import { CountdownVariants } from '../../components/Countdown/types';
+
+export interface BidSnapshotProps extends ComponentProps<'div'> {
+  /**
+   * Active bid of current user. -  '$1,000'
+   */
+  activeBid?: string;
+  /**
+   * State of the object
+   */
+  auctionStatus?: AuctionStatus;
+  /**
+   * An array of bids for the Object. Should include starting bid as first element and current bid as last element
+   */
+  bids?: string[];
+  /**
+   * Bids label text, a fucntion for label of bids amoutn (2 bids, 3 bids, etc) where the number is the length of the bids array.
+   */
+  bidsLabelText?: (numberOfBids: number) => string;
+  /**
+   * Current bid text, a string for label of current bid detail
+   */
+  currentBidText?: string;
+  /**
+   * Closing label text, for countdown timer
+   */
+  closingText?: string;
+  /**
+   * Language selection for the application
+   */
+  lang?: keyof typeof SupportedLanguages;
+  /**
+   * End time for this object
+   */
+  lotCloseDate?: Date;
+  /**
+   * Starting bid text, a string for label of starting bid detail
+   */
+  startingBidText?: string;
+  /**
+   * Sold for label text, a string for label of sold for detail
+   */
+  soldForText?: string;
+  /**
+   * Won for label text, a string for label of won for detail
+   */
+  wonForText?: string;
+}
+
+const bidsTranslation = (numberOfBids: number) => (numberOfBids === 1 ? `${numberOfBids} bid` : `${numberOfBids} bids`);
+/**
+ * ## Overview
+ *
+ * Overview of this widget
+ *
+ * [Figma Link](https://www.figma.com/design/hMu9IWH5N3KamJy8tLFdyV/EASEL-Compendium%3A-Tokens%2C-Components-%26-Patterns?node-id=16780-34969&node-type=instance&m=dev)
+ *
+ * [Storybook Link](https://phillips-seldon.netlify.app//?path=/docs/patterns-bidsnapshot--overview)
+ */
+const BidSnapshot = forwardRef<HTMLDivElement, BidSnapshotProps>(
+  (
+    {
+      activeBid,
+      auctionStatus = AuctionStatus.ready,
+      bids = [],
+      bidsLabelText = bidsTranslation,
+      children,
+      className,
+      closingText = 'Closes in',
+      currentBidText = 'Current bid',
+      lang = 'en',
+      lotCloseDate,
+      startingBidText = 'Starting bid',
+      soldForText = 'Sold for',
+      wonForText = 'Won for',
+      ...props
+    },
+    ref,
+  ) => {
+    const { className: baseClassName, ...commonProps } = getCommonProps(props, 'BidSnapshot');
+    const currentDate = new Date();
+    return (
+      <div {...commonProps} className={classnames(baseClassName, className)} {...props} ref={ref}>
+        {
+          // auctionStatus is not past (auction is active or upcoming),
+          !(auctionStatus === AuctionStatus.past) ? (
```

Comment:
can we avoid nested conditions?

I think cleaner way to create more single conditions like `condition ? <Component/> : null`
pretty sure it will be easier to read

Reply:
or we can move them to variables

Reply:
agree that nested ternaries can get freaky